File: two_sum.js
// 两数之和
// https://leetcode-cn.com/problems/two-sum/description/

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  let dict = {}

  for (let i in nums) {
    let num = nums[i]

    if (num in dict) {
      return [dict[num], i]
    }

    dict[target - num] = i
  }
}

File: two_sum.swift
// 两数之和
// https://leetcode-cn.com/problems/two-sum/description/

class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var dict: [Int: Int] = [:]

        for i in 0..<nums.count {
            let num = nums[i]

            if dict.keys.contains(num) {
                return [dict[num]!, i]
            }

            dict[target - num] = i
        }
        return []
    }
}
